utils/bitarray_utils.py

```diff
@@ -4,8 +4,16 @@
 from typing import Tuple
 
 
-def get_bit_width(size) -> int:
-    return int(np.ceil(np.log2(size)))
+def get_bit_width(x) -> int:
```

Comment:
TODO - add test for this, checking for boundary values like powers of 2, powers of 2 minus 1 (looks correct to me but I have made mistakes in the past, so better to add simple tests)
also add assertion for positivity
do we want to support x=0

Reply:
good point, I think there is indeed a mistake here! 
if x = 4, this returns 2, which is not enough :-|

Reply:
fixed + added tests